docs/docsite/rst/dev_guide/developing_modules_in_bulk.rst

```diff
@@ -0,0 +1,160 @@
+Partner information for submitting modules
+==========================================
+
+.. contents:: Topics
+
+.. _module_dev_welcome:
+
+Welcome
+```````
+This section discusses some of the specifics around getting multiple related modules into Ansible.
+
+The information in the page is useful for both companies wishing to add modules for their own products, as well as for users of 3rd party products wishing to add Ansible support.
+
+It's based on the knowledge on the Core Team and Community on what we've found is the best practice for the module developer(s), the Community, and the Ansible Core Team.
+
+
+
+Before you start coding
+```````````````````````
+
+Although it's tempting to get straight into coding, there are a few things to be aware of first. This list has been drawn to ensure the best modules are developed and to speed up the review process, which means your modules get accepted into Ansible with minimal fuss.
+
+* Read though all the pages linked off :doc:`developing_modules`, paying particular focus to the :doc:`developing_modules_checklist`.
+* For new modules going into Ansible 2.4 we are raising the bar so they must pass `pep8 --ignore=E402 --max-line-length=160` cleanly.
+* All new modules going into Ansible 2.3 must support Python 2.4+ and Python 3.5+. For Ansible 2.4 the requirement will change to Python and Python 3.5+. If this is an issues mention it in the "Speak to us" section later in this document.
+* Understand that all modules shipped with Ansible MUST be done so under the GPLv3 license. Files under the `lib/ansible/module_utils/` directory must be under the FIXME license.
```

Comment:
for module_utils we recommend BSD so others can freely make and distribute modules based on the shared code,  but we do allow GPL also in case submitter does not want this.